sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTable.scala

```diff
@@ -106,6 +106,18 @@ class InMemoryTable(
           case (micros: Long, TimestampType) =>
             val localDate = DateTimeUtils.microsToInstant(micros).atZone(UTC).toLocalDate
             ChronoUnit.YEARS.between(EPOCH_LOCAL_DATE, localDate)
+          case _ => throw new Exception("""match may not be exhaustive.
```

Comment:
I'd make this more meaningful, like an IllegalArgumentException referring to the thing that does not match. Something simple.